Apk per ABI fails with TargetFrameworkVersion v7.1 and lower when Aapt2 is enabled

### Steps to Reproduce

1. ``` msbuild /t:SignAndroidPackage *.csproj /p:Configuration=Release /p:TargetFrameworkVersion=v7.1 /p:AndroidCreatePackagePerAbi=True /p:AndroidUseAapt2=True```

This could be considered a regression in d16-2 as `Aapt2` resource processing is now enabled by default.

### Actual Behavior
I've noticied a difference in `Aapt2Link` task behavior when comparing good and bad builds (both attached below). For some reason ABI specific manifest files don't appear to be processed by `Aapt2Link` when targeting a lower API level, and the ABI specific resource files are never created:

```
error MSB4018: The "BuildApk" task failed unexpectedly.
error MSB4018: System.IO.FileNotFoundException: Could not find file 'obj\Release\71\android\bin\packaged_resources-armeabi-v7a'
```


### Version Information
XA D16-2 9.4.0.4 (HEAD/cc0e8b8d5)

<!--
1. On macOS and within Visual Studio, select Visual Studio > About Visual Studio, then click the Show Details button, then click the Copy Information button.

2. Paste below this comment block.
-->

### Log File
[good-bad-binlogs.zip](https://github.com/xamarin/xamarin-android/files/3158319/good-bad-binlogs.zip)

<!--
1. On macOS and within Visual Studio:
    a. Click **Tools** > **SDK Command Prompt**.
    b. Within the launched `Terminal.app` window, run:

            adb logcat -d | pbcopy

2. Paste below this comment block
-->



<!--
Switch to the "Preview" tab to ensure your issue renders correctly.

Please add an appropriate "Area:" label in the Labels section.
-->

--- a/src/Xamarin.Android.Build.Tasks/Tasks/Aapt2.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tasks/Aapt2.cs
@@ -69,7 +69,7 @@ namespace Xamarin.Android.Tasks {
 				proc.ErrorDataReceived += (sender, e) => {
 					if (e.Data != null)
 						lock (lockObject)
-							output.Add (new OutputLine (e.Data, stdError: true));
+							output.Add (new OutputLine (e.Data, stdError: !IsAapt2Warning (e.Data)));
 					else
 						stderr_completed.Set ();
 				};
@@ -91,6 +91,23 @@ namespace Xamarin.Android.Tasks {
 					stdout_completed.WaitOne (TimeSpan.FromSeconds (30));
 				return proc.ExitCode == 0 && !output.Any (x => x.StdError);
 			}
+		}
+
+		bool IsAapt2Warning (string singleLine)
+		{
+			var match = AndroidToolTask.AndroidErrorRegex.Match (singleLine.Trim ());
+			if (match.Success) {
+				var file = match.Groups ["file"].Value;
+				var level = match.Groups ["level"].Value.ToLowerInvariant ();
+				var message = match.Groups ["message"].Value;
+				if (file.StartsWith ("W/", StringComparison.OrdinalIgnoreCase))
+					return true;
+				if (message.Contains ("warn:"))
+					return true;
+				if (level.Contains ("warning"))
+					return true;
+			}
+			return false;
 		}
 
 		protected bool LogAapt2EventsFromOutput (string singleLine, MessageImportance messageImportance, bool apptResult)

--- a/src/Xamarin.Android.Build.Tasks/Tasks/Aapt2Compile.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tasks/Aapt2Compile.cs
@@ -66,12 +66,8 @@ namespace Xamarin.Android.Tasks {
 					archives.Add (new TaskItem (outputArchive));
 			}
 			foreach (var line in output) {
-				if (line.StdError) {
-					if (!LogAapt2EventsFromOutput (line.Line, MessageImportance.Normal, success))
-						break;
-				} else {
-					LogMessage (line.Line, MessageImportance.Normal);
-				}
+				if (!LogAapt2EventsFromOutput (line.Line, MessageImportance.Normal, success))
+					break;
 			}
 		}
 

--- a/src/Xamarin.Android.Build.Tasks/Tasks/Aapt2Link.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tasks/Aapt2Link.cs
@@ -222,12 +222,8 @@ namespace Xamarin.Android.Tasks {
 				? File.Exists (Path.Combine (currentResourceOutputFile + ".bk"))
 				: ret;
 			foreach (var line in output) {
-				if (line.StdError) {
-					if (!LogAapt2EventsFromOutput (line.Line, MessageImportance.Normal, success))
-						break;
-				} else {
-					LogMessage (line.Line, MessageImportance.Normal);
-				}
+				if (!LogAapt2EventsFromOutput (line.Line, MessageImportance.Normal, success))
+					break;
 			}
 			if (ret && !string.IsNullOrEmpty (currentResourceOutputFile)) {
 				var tmpfile = currentResourceOutputFile + ".bk";

--- a/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/PackagingTest.cs
+++ b/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/PackagingTest.cs
@@ -371,6 +371,41 @@ string.Join ("\n", packages.Select (x => metaDataTemplate.Replace ("%", x.Id))) 
 		}
 
 		[Test]
+		public void CheckAapt2WarningsDoNotGenerateErrors ()
+		{
+			//https://github.com/xamarin/xamarin-android/issues/3083
+			var proj = new XamarinAndroidApplicationProject () {
+				IsRelease = true,
+				TargetFrameworkVersion = "v7.1",
+				UseLatestPlatformSdk = false,
+			};
+			proj.PackageReferences.Add (KnownPackages.XamarinForms_2_3_4_231);
+			proj.PackageReferences.Add (KnownPackages.AndroidSupportV4_25_4_0_1);
+			proj.PackageReferences.Add (KnownPackages.SupportCompat_25_4_0_1);
+			proj.PackageReferences.Add (KnownPackages.SupportCoreUI_25_4_0_1);
+			proj.PackageReferences.Add (KnownPackages.SupportCoreUtils_25_4_0_1);
+			proj.PackageReferences.Add (KnownPackages.SupportDesign_25_4_0_1);
+			proj.PackageReferences.Add (KnownPackages.SupportFragment_25_4_0_1);
+			proj.PackageReferences.Add (KnownPackages.SupportMediaCompat_25_4_0_1);
+			proj.PackageReferences.Add (KnownPackages.SupportV7AppCompat_25_4_0_1);
+			proj.PackageReferences.Add (KnownPackages.SupportV7CardView_25_4_0_1);
+			proj.PackageReferences.Add (KnownPackages.SupportV7MediaRouter_25_4_0_1);
+			proj.SetProperty (proj.ReleaseProperties, KnownProperties.AndroidCreatePackagePerAbi, true);
+			proj.SetProperty (proj.ReleaseProperties, KnownProperties.AndroidSupportedAbis, "armeabi-v7a;x86");
+			using (var b = CreateApkBuilder (Path.Combine ("temp", TestName))) {
+				if (!b.TargetFrameworkExists (proj.TargetFrameworkVersion))
+					Assert.Ignore ($"Skipped as {proj.TargetFrameworkVersion} not available.");
+				Assert.IsTrue (b.Build (proj), "first build should have succeeded.");
+				var packagedResource = Path.Combine (b.Root, b.ProjectDirectory, proj.IntermediateOutputPath, "android", "bin", "packaged_resources");
+				FileAssert.Exists (packagedResource, $"{packagedResource} should have been created.");
+				var packagedResourcearm = packagedResource + "-armeabi-v7a";
+				FileAssert.Exists (packagedResourcearm, $"{packagedResourcearm} should have been created.");
+				var packagedResourcex86 = packagedResource + "-x86";
+				FileAssert.Exists (packagedResourcex86, $"{packagedResourcex86} should have been created.");
+			}
+		}
+
+		[Test]
 		public void CheckAppBundle ([Values (true, false)] bool isRelease)
 		{
 			var proj = new XamarinAndroidApplicationProject () {

